src/ucp/wireup/wireup.c

```diff
@@ -825,7 +801,25 @@ ucp_wireup_connect_lane_to_iface(ucp_ep_h ep, ucp_lane_index_t lane,
         return status;
     }
 
-    ucp_wireup_assign_lane(ep, lane, uct_ep, "");
+    /* If EP already exists, it's a wireup proxy (only client-server),
+     * and we need to update its next_ep instead of replacing it.
+     */
+    if (ep->uct_eps[lane] == NULL) {
+        ucs_trace("ep %p: assign uct_ep[%d]=%p", ep, lane, uct_ep);
+        ep->uct_eps[lane] = uct_ep;
+    } else {
+        ucs_assert(ucp_wireup_ep_test(ep->uct_eps[lane]) &&
+                   (ucp_ep_has_cm_lane(ep) || !ucp_ep_is_sockaddr_stub(ep)));
```

Comment:
1. will ucp_ep_is_sockaddr_stub also be removed when old sockaddr is removed? add comment
2. can we split to 2 asserts instead of &&, will make failure more clear

Reply:
let's remove the assert for old sockaddr